src/trace_processor/importers/proto/track_event_parser.cc

```diff
@@ -363,11 +369,17 @@ void TrackEventParser::ParseChromeProcessDescriptor(
 }
 
 UniqueTid TrackEventParser::ParseThreadDescriptor(
-    protozero::ConstBytes thread_descriptor) {
+    protozero::ConstBytes thread_descriptor,
+    bool is_sandboxed) {
   protos::pbzero::ThreadDescriptor::Decoder decoder(thread_descriptor);
-  UniqueTid utid = context_->process_tracker->UpdateThread(
-      static_cast<uint32_t>(decoder.tid()),
-      static_cast<uint32_t>(decoder.pid()));
+  auto pid = static_cast<int64_t>(decoder.pid());
+  auto tid = static_cast<int64_t>(decoder.tid());
+  // If tid is sandboxed then use a unique synthetic tid, to avoid
+  // having concurrent threads with the same tid.
+  if (is_sandboxed) {
+    tid = ProcessTracker::CreateSyntheticTid(tid, pid);
```

Comment:
Won't this create conflicting tids between parsing and tokenizing this descriptor?

I'm not quite sure what the consequence of that is (we lack some comments about why we UpdateThread() below again), but I wonder if we need to keep track of the tokenization-time translations into synthetic tids somehow?

Also not sure if we need to do something about two different tracks referencing the same thread. I recall that we had this in the past (e.g. one thread emitting events on behalf of another) but I think we got rid of that at some point in Chrome.